recipe/meta.yaml

```diff
@@ -1,38 +1,39 @@
-{% set version = "4.1.3" %}
+{% set version = "5.0.5" %}
 
 package:
   name: moto
   version: {{ version }}
 
 source:
   - url: https://pypi.io/packages/source/m/moto/moto-{{ version }}.tar.gz
-    sha256: c8200ccaa9440c2e9daa0bd5e0bd768a719db5a2c82ea8d782f0e3fa09a3c5e2
+    sha256: 2eaca2df7758f6868df420bf0725cd0b93d98709606f1fb8b2343b5bdc822d91
 
 build:
   number: 0
   # there seem to be missing docker-py >=2.5.1 and cfn-lint >=0.4.0 for s390x
-  skip: True  # [py<37 or s390x]
+  # there is also and issue with 3.12 and graphql-core which results in a failed build
+  skip: True  # [py<37 or s390x or py>311]
   entry_points:
     - moto_server = moto.server:main
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
   host:
     - python
     - pip
-    - setuptools
+    - setuptools >=40.6.0
     - wheel
   run:
     - python
     # we need to bound this since 198 changed some of the api
     - boto3 >=1.9.201
-    - botocore >=1.12.201
+    - botocore >=1.14.0
     - cryptography >=3.3.1
     - requests >=2.5
     - xmltodict
     - werkzeug >=0.5,!=2.2.0,!=2.2.1
     - python-dateutil >=2.1,<3.0.0
-    - responses >=0.13.0
+    - responses >=0.15.0
     - jinja2 >=2.10.1
```

Comment:
We should identify which optional group was made required, verify that all its dependencies are still in run. And put other optional dependencies under run_constrained.

Reply:
It seems to me that we had the dependencies for server (given the entry point is there).  They are also almost identical to proxy. So we should support both. https://github.com/getmoto/moto/blob/5.0.5/setup.cfg#L76 and https://github.com/getmoto/moto/blob/5.0.5/setup.cfg#L61

In that case that would mean:
- removing sshpubkeys (not in current deps)
- moving antlr4-python3-runtime and pyparsing >=3.0.7 to run
- adding jsonpath-ng, py-partiql-parser==0.5.4, setuptools, multipart
- add entrypoint `moto_proxy = moto.proxy:main`